src/main/java/com/eprosima/fastrtps/idl/templates/RTPSPubSubTypeSource.stg

```diff
@@ -68,7 +68,7 @@ typedef_decl(ctx, parent, typedefs) ::= <<>>
 struct_type(ctx, parent, struct) ::= <<
 $if(parent.IsInterface)$$parent.name$_$endif$$struct.name$PubSubType::$if(parent.IsInterface)$$parent.name$_$endif$$struct.name$PubSubType()
 {
-    setName("$struct.scopedname$");
+    $if(ctx.GenerateTypesROS2)$    setName("$struct.ROS2Scopedname$");$else$    setName("$struct.scopedname$");$endif$
```

Comment:
Is the part `$struct.ROS2Scopedname$'  working for you? If I remember well, calling `struct.scopedname` will invoke in java to `struct.getScopedname()`. Then, to getting the ros2 scopedname it should be `struct.rOS2Scopedname()`. But if you check this is working, I agree with it.

Reply:
Yes it is working. `struct.ROSScopedname` will invoke getROS2Scopedname. It doesn't seem to be case sentitive in the first letter.